codechain/primitives/AssetAddress.py

```diff
@@ -0,0 +1,110 @@
+import re
+from typing import List
+from typing import NamedTuple
+from typing import Tuple
+from typing import Union
+
+from ..crypto import bech32_decode
+from ..crypto import bech32_encode
+from .HexString import H160
+
+
+class MultiSig(NamedTuple):
+    n: int
+    m: int
+    pubkeys: List[H160]
+
+
+class AssetAddress:
```

Comment:
It is useful to have `__hash__` and `__eq__` method. It allows `AssetAddress` to be used as a dictionary key. It would've been better to make it a value type, but it is quite difficult to express in Python for this type.
https://stackoverflow.com/questions/4901815/object-of-custom-type-as-dictionary-key

Reply:
I added `__hash__` method.